### PR TITLE
feat(papyrus_base_layer): start implementing eth events api

### PR DESCRIPTION
**Pull request by giladchase**
- Currently only implementing the simplest event to parse, the rest are coming up.
- removed two unused error variants: `FeltParseError`, `Serde`

### CODE REVIEW

**Review by elintul — APPROVED** [review]
> Reviewed 1 of 6 files at r1, 5 of 5 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2845)* status: all files reviewed, 3 unresolved discussions (waiting on @giladchase and @matan-starkware)

___
*[`crates/papyrus_base_layer/src/ethereum_base_layer_contract.rs` line 120 at r2](https://reviewable.io/reviews/starkware-libs/sequencer/2845#-OEh_ibO4HI2Ah7Bhq6l:-OEh_ibO4HI2Ah7Bhq6m:b-tjrazg) ([raw file](https://github.com/starkware-libs/sequencer/blob/9b7ce5ddf279498519c3d95ae98bdce22715f779/crates/papyrus_base_layer/src/ethereum_base_layer_contract.rs#L120)):*
> ```XML
>     StarknetApiParsingError(StarknetApiError),
>     #[error("{0:?}")]
>     UnhandledL1Event(alloy_primitives::Log),
> ```

Can you shortly explain the scenario in which this variant is returned?



_Code quote:_
```XML
UnhandledL1Event(alloy_primitives::Log)
```
___
*[`crates/papyrus_base_layer/src/eth_events.rs` line 76 at r2](https://reviewable.io/reviews/starkware-libs/sequencer/2845#-OEh_Lu6Dk4VWkqfjmhy:-OEh_Lu6Dk4VWkqfjmhz:b-3bttee) ([raw file](https://github.com/starkware-libs/sequencer/blob/9b7ce5ddf279498519c3d95ae98bdce22715f779/crates/papyrus_base_layer/src/eth_events.rs#L76)):*
> ```XML
>     })
> }
> ```

Don't we have the utils below somewhere in the codebase?
Add a TODO to reuse if you prefer not to find/move them now.

**Review by elintul — APPROVED** [review]
> Reviewed 4 of 4 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2845)* status: all files reviewed, 3 unresolved discussions (waiting on @giladchase and @matan-starkware)

**Review by giladchase — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2845)* status: all files reviewed, 3 unresolved discussions (waiting on @elintul, @graphite-app[bot], and @matan-starkware)

___
*[`crates/papyrus_base_layer/src/eth_events.rs` line 76 at r2](https://reviewable.io/reviews/starkware-libs/sequencer/2845#-OEh_Lu6Dk4VWkqfjmhy:-OEszME6D7Elu6VH5gky:b-tszzfk) ([raw file](https://github.com/starkware-libs/sequencer/blob/9b7ce5ddf279498519c3d95ae98bdce22715f779/crates/papyrus_base_layer/src/eth_events.rs#L76)):*
<details><summary><i>Previously, elintul (Elin) wrote…</i></summary><blockquote>

Don't we have the utils below somewhere in the codebase?
Add a TODO to reuse if you prefer not to find/move them now.
</blockquote></details>

It's cause `EthereumContractAddress` and `U256` are types from the `alloy` crate for communicating with L1, which probably shouldn't be a dependency anywhere else outside this module, I think
___
*[`crates/papyrus_base_layer/src/ethereum_base_layer_contract.rs` line 120 at r2](https://reviewable.io/reviews/starkware-libs/sequencer/2845#-OEh_ibO4HI2Ah7Bhq6l:-OEszaSE5QGpk1LqTCFB:b-c4rep3) ([raw file](https://github.com/starkware-libs/sequencer/blob/9b7ce5ddf279498519c3d95ae98bdce22715f779/crates/papyrus_base_layer/src/ethereum_base_layer_contract.rs#L120)):*
<details><summary><i>Previously, elintul (Elin) wrote…</i></summary><blockquote>

Can you shortly explain the scenario in which this variant is returned?
</blockquote></details>

< Hofer >
Maybe i should have used `unimplemented!()`, but it's a bit aggresive.
Basically if someone calls EthereumBaseLayer's `events` with the arg `event_identifiers` including legal Starknet L1 events that are not the four events I'm currently supporting (L1Message, the 2 cancelation events and the consumed event), the match in eth_events.rs:25 above will get to the Unhandled branch.

That is, at some point this base_layer module will support all StarknetEvents, then I can remove this variant because the match will cover all options.

I wasn't sure about returning error or calling unimplemnted! because technically the user gave legal input (so returning error isn't cool), but it simply is not supported yet (which typically calls for unimplemented!).
< /Hofer >

WDYT?

**Review by elintul — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2845)* status: all files reviewed, 2 unresolved discussions (waiting on @giladchase, @graphite-app[bot], and @matan-starkware)

___
*[`crates/papyrus_base_layer/src/ethereum_base_layer_contract.rs` line 120 at r2](https://reviewable.io/reviews/starkware-libs/sequencer/2845#-OEh_ibO4HI2Ah7Bhq6l:-OEt9ITW-d22QRVLqD-o:b-7p8hnw) ([raw file](https://github.com/starkware-libs/sequencer/blob/9b7ce5ddf279498519c3d95ae98bdce22715f779/crates/papyrus_base_layer/src/ethereum_base_layer_contract.rs#L120)):*
<details><summary><i>Previously, giladchase wrote…</i></summary><blockquote>

< Hofer >
Maybe i should have used `unimplemented!()`, but it's a bit aggresive.
Basically if someone calls EthereumBaseLayer's `events` with the arg `event_identifiers` including legal Starknet L1 events that are not the four events I'm currently supporting (L1Message, the 2 cancelation events and the consumed event), the match in eth_events.rs:25 above will get to the Unhandled branch.

That is, at some point this base_layer module will support all StarknetEvents, then I can remove this variant because the match will cover all options.

I wasn't sure about returning error or calling unimplemnted! because technically the user gave legal input (so returning error isn't cool), but it simply is not supported yet (which typically calls for unimplemented!).
< /Hofer >

WDYT?
</blockquote></details>

Go with the option that won't get forgotten longterm. :sweat_smile:

**Review by elintul — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2845#-:-OEt9QCc76ickG_i6eGq:bnfp4nl)* status: all files reviewed, 2 unresolved discussions (waiting on @giladchase, @graphite-app[bot], and @matan-starkware)

**Review by giladchase — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2845)* status: all files reviewed, 2 unresolved discussions (waiting on @elintul, @graphite-app[bot], and @matan-starkware)

___
*[`crates/papyrus_base_layer/src/ethereum_base_layer_contract.rs` line 120 at r2](https://reviewable.io/reviews/starkware-libs/sequencer/2845#-OEh_ibO4HI2Ah7Bhq6l:-OEwj1QrBZ7xnmcu62E-:b-3ma6d8) ([raw file](https://github.com/starkware-libs/sequencer/blob/9b7ce5ddf279498519c3d95ae98bdce22715f779/crates/papyrus_base_layer/src/ethereum_base_layer_contract.rs#L120)):*
<details><summary><i>Previously, elintul (Elin) wrote…</i></summary><blockquote>

Go with the option that won't get forgotten longterm. :sweat_smile:
</blockquote></details>

Oh neither would, someone will hit either an error or a panic when they try to add support for a new event.
I'll just keep it as an error then.

**Review by matan-starkware — APPROVED** [review]
> Reviewed 1 of 6 files at r1, 2 of 5 files at r2, 4 of 4 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2845)* status: all files reviewed, 4 unresolved discussions (waiting on @giladchase and @graphite-app[bot])

___
*[`crates/papyrus_base_layer/src/ethereum_base_layer_contract.rs` line 101 at r3](https://reviewable.io/reviews/starkware-libs/sequencer/2845#-OEwmI2r6jb-8xtUc8su:-OEwmI2sERIpNzbu_d55:b-ylovn5) ([raw file](https://github.com/starkware-libs/sequencer/blob/d65482b6a4a26ec83a8e3f7316c15578ce62bfdf/crates/papyrus_base_layer/src/ethereum_base_layer_contract.rs#L101)):*
> ```XML
>             .into_iter()
>             .map(TryInto::try_into)
>             .collect()
> ```

Can you explain to me how this call works?



_Code quote:_
```XML
        // Get all logs from the latest block that match the filter.
        self.contract
            .provider()
            .get_logs(&filter)
            .await?
            .into_iter()
            .map(TryInto::try_into)
            .collect()
```
___
*[`crates/papyrus_base_layer/src/eth_events.rs` line 55 at r3](https://reviewable.io/reviews/starkware-libs/sequencer/2845#-OEtx8OBDH8B6GUNiMRa:-OEtx8OBDH8B6GUNiMRb:b641gvn) ([raw file](https://github.com/starkware-libs/sequencer/blob/d65482b6a4a26ec83a8e3f7316c15578ce62bfdf/crates/papyrus_base_layer/src/eth_events.rs#L55)):*
> ```XML
>         )
>     }
> }
> ```

So this is here because we don't directly have this type in rust? I saw the `sol!` macro.



_Code quote:_
```XML
impl TryFrom<Starknet::MessageToL2Canceled> for EventData {
    type Error = EthereumBaseLayerError;

    fn try_from(event: Starknet::MessageToL2Canceled) -> EthereumBaseLayerResult<Self> {
        create_l1_event_data(
            event.fromAddress,
            event.toAddress,
            event.selector,
            &event.payload,
            event.nonce,
        )
    }
}
```
___
*[`crates/papyrus_base_layer/src/eth_events.rs` line 79 at r3](https://reviewable.io/reviews/starkware-libs/sequencer/2845#-OEwlQBa9IbzLccnx2Ek:-OEwlQBa9IbzLccnx2El:b-59l80h) ([raw file](https://github.com/starkware-libs/sequencer/blob/d65482b6a4a26ec83a8e3f7316c15578ce62bfdf/crates/papyrus_base_layer/src/eth_events.rs#L79)):*
> ```XML
> pub fn felt_from_eth_address(address: EthereumContractAddress) -> Felt {
>     Felt::from_bytes_be_slice(address.0.as_slice())
> }
> ```

Are we going to use this in other places? If not can we just inline this?



_Code quote:_
```XML
pub fn felt_from_eth_address(address: EthereumContractAddress) -> Felt {
    Felt::from_bytes_be_slice(address.0.as_slice())
}
```

**Review by elintul — APPROVED** [review]
> Reviewed 3 of 3 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2845)* status: all files reviewed, 4 unresolved discussions (waiting on @giladchase and @graphite-app[bot])

**Review by giladchase — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2845)* status: all files reviewed, 4 unresolved discussions (waiting on @graphite-app[bot] and @matan-starkware)

___
*[`crates/papyrus_base_layer/src/eth_events.rs` line 55 at r3](https://reviewable.io/reviews/starkware-libs/sequencer/2845#-OEtx8OBDH8B6GUNiMRa:-OEwm_i3Ad-CjbuvJMXN:bcfkqfw) ([raw file](https://github.com/starkware-libs/sequencer/blob/d65482b6a4a26ec83a8e3f7316c15578ce62bfdf/crates/papyrus_base_layer/src/eth_events.rs#L55)):*
<details><summary><i>Previously, matan-starkware wrote…</i></summary><blockquote>

So this is here because we don't directly have this type in rust? I saw the `sol!` macro.
</blockquote></details>

We have it, `Starknet::MessageToL2Canceled` is generated from `sol!` and has all the fields seen here.
Thing is, I can't leak that type out of this crate, because users should all use the BaseLayerContract trait, which is agnostic to ethereum and to `alloy` in particular. Therefore, I can't export `sol!` types, since they derive from alloy :pensive: .

Another alternative could have been to newType or re-export them, but for now I wanted to do the simplest solution possible in order to push this through.
___
*[`crates/papyrus_base_layer/src/eth_events.rs` line 79 at r3](https://reviewable.io/reviews/starkware-libs/sequencer/2845#-OEwlQBa9IbzLccnx2Ek:-OEwoXuB4_1Ld9EdVt3U:b2xeyok) ([raw file](https://github.com/starkware-libs/sequencer/blob/d65482b6a4a26ec83a8e3f7316c15578ce62bfdf/crates/papyrus_base_layer/src/eth_events.rs#L79)):*
<details><summary><i>Previously, matan-starkware wrote…</i></summary><blockquote>

Are we going to use this in other places? If not can we just inline this?
</blockquote></details>

Thanks! Inlined.
It had previous uses previously, but they went away during the review.
___
*[`crates/papyrus_base_layer/src/ethereum_base_layer_contract.rs` line 101 at r3](https://reviewable.io/reviews/starkware-libs/sequencer/2845#-OEwmI2r6jb-8xtUc8su:-OEwof9MBFe_tKrBeX0U:bol0fsd) ([raw file](https://github.com/starkware-libs/sequencer/blob/d65482b6a4a26ec83a8e3f7316c15578ce62bfdf/crates/papyrus_base_layer/src/ethereum_base_layer_contract.rs#L101)):*
<details><summary><i>Previously, matan-starkware wrote…</i></summary><blockquote>

Can you explain to me how this call works?
</blockquote></details>

Sure, i split it into two lines to improve readability.

The first line gets `provider()`, which is the actual L1 client that was initialized through `alloy`.
Then calls `get_logs(filter)`, which is an `alloy` API (from its L1 client) that fetches all L1 events that match the given filter, which for us is `{from,to}_block` and a whitelist of events to fetch, where the white list is a list of &str such that each `&str` is `keccak(<signature_of_event_in_l1>)`, so for example: `keccak("LogMessageToL1(uint256, address, uint256[])")`.

The second line casts the vector of events from the first line, which all have the inner `alloy` event type called `Log`, into the `Sequencer` event  type we called `Eventdata`, using the casts from `eth_events.rs`.

Since these cast can throw an error, we then cast the vector of results into a result<vector> [using collect](https://doc.rust-lang.org/rust-by-example/error/iter_result.html#fail-the-entire-operation-with-collect).

**Review by matan-starkware — CHANGES_REQUESTED** [review]
> Reviewed 3 of 3 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2845)* status: all files reviewed, 4 unresolved discussions (waiting on @giladchase and @graphite-app[bot])

___
*[`crates/papyrus_base_layer/src/ethereum_base_layer_contract.rs` line 101 at r3](https://reviewable.io/reviews/starkware-libs/sequencer/2845#-OEwmI2r6jb-8xtUc8su:-OEwyoNj2wlO3ummKIqB:b-zcjvte) ([raw file](https://github.com/starkware-libs/sequencer/blob/d65482b6a4a26ec83a8e3f7316c15578ce62bfdf/crates/papyrus_base_layer/src/ethereum_base_layer_contract.rs#L101)):*
<details><summary><i>Previously, giladchase wrote…</i></summary><blockquote>

Sure, i split it into two lines to improve readability.

The first line gets `provider()`, which is the actual L1 client that was initialized through `alloy`.
Then calls `get_logs(filter)`, which is an `alloy` API (from its L1 client) that fetches all L1 events that match the given filter, which for us is `{from,to}_block` and a whitelist of events to fetch, where the white list is a list of &str such that each `&str` is `keccak(<signature_of_event_in_l1>)`, so for example: `keccak("LogMessageToL1(uint256, address, uint256[])")`.

The second line casts the vector of events from the first line, which all have the inner `alloy` event type called `Log`, into the `Sequencer` event  type we called `Eventdata`, using the casts from `eth_events.rs`.

Since these cast can throw an error, we then cast the vector of results into a result<vector> [using collect](https://doc.rust-lang.org/rust-by-example/error/iter_result.html#fail-the-entire-operation-with-collect).
</blockquote></details>

Thanks
___
*[`crates/papyrus_base_layer/src/ethereum_base_layer_contract.rs` line 94 at r4](https://reviewable.io/reviews/starkware-libs/sequencer/2845#-OEwxkpF7QVzEhHejoZz:-OEwxIpH-xHOMnydZ6H1:bnpj9ll) ([raw file](https://github.com/starkware-libs/sequencer/blob/8d13ce882fac7aa074a12081b03d1070633f4694/crates/papyrus_base_layer/src/ethereum_base_layer_contract.rs#L94)):*
> ```XML
>             .to_block(until_block);
> 
>         // Get all logs from the latest block that match the filter.
> ```

This is a little confusing, since the trait has the comment
```
    /// Get specific events from the Starknet base contract between two L1 block numbers.
```

That comment makes it sound like we get events for all blocks in the range provided. The comment here makes it sound like we only get events from that last block in the range provided.



_Suggestion:_
```XML
        // Get all logs from the latest block that matches the filter.
```
___
*[`crates/papyrus_base_layer/src/ethereum_base_layer_contract.rs` line 95 at r4](https://reviewable.io/reviews/starkware-libs/sequencer/2845#-OEwxVMm9s9fENPwMq-F:-OEwxVMn7xMX0Wv-sqkl:b3kre) ([raw file](https://github.com/starkware-libs/sequencer/blob/8d13ce882fac7aa074a12081b03d1070633f4694/crates/papyrus_base_layer/src/ethereum_base_layer_contract.rs#L95)):*
> ```XML
> 
>         // Get all logs from the latest block that match the filter.
>         let l1_events_that_match_the_filter = self.contract.provider().get_logs(&filter).await?;
> ```

?



_Suggestion:_
```XML
        let matching_logs = self.contract.provider().get_logs(&filter).await?;
```
___
*[`crates/papyrus_base_layer/src/ethereum_base_layer_contract.rs` line 96 at r4](https://reviewable.io/reviews/starkware-libs/sequencer/2845#-OEwySC4C83GIQENYvEo:-OEwySC4C83GIQENYvEp:b-kihm8o) ([raw file](https://github.com/starkware-libs/sequencer/blob/8d13ce882fac7aa074a12081b03d1070633f4694/crates/papyrus_base_layer/src/ethereum_base_layer_contract.rs#L96)):*
> ```XML
>         // Get all logs from the latest block that match the filter.
>         let l1_events_that_match_the_filter = self.contract.provider().get_logs(&filter).await?;
>         // Try to convert all events to L1Events and return an error if one of them failed.
> ```

Mention the try and error stuff it basic rust (we can see TryInto).



_Suggestion:_
```XML
        // Convert logs to L1Events.
```

**Review by giladchase — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2845)* status: 6 of 7 files reviewed, 3 unresolved discussions (waiting on @elintul, @graphite-app[bot], and @matan-starkware)

___
*[`crates/papyrus_base_layer/src/ethereum_base_layer_contract.rs` line 94 at r4](https://reviewable.io/reviews/starkware-libs/sequencer/2845#-OEwxkpF7QVzEhHejoZz:-OExU7jyDATcr_V4AUFy:b4t21th) ([raw file](https://github.com/starkware-libs/sequencer/blob/8d13ce882fac7aa074a12081b03d1070633f4694/crates/papyrus_base_layer/src/ethereum_base_layer_contract.rs#L94)):*
<details><summary><i>Previously, matan-starkware wrote…</i></summary><blockquote>

This is a little confusing, since the trait has the comment
```
    /// Get specific events from the Starknet base contract between two L1 block numbers.
```

That comment makes it sound like we get events for all blocks in the range provided. The comment here makes it sound like we only get events from that last block in the range provided.
</blockquote></details>

Ya the comment isn't right, removed, the comment in the trait is sufficient I think.
___
*[`crates/papyrus_base_layer/src/ethereum_base_layer_contract.rs` line 96 at r4](https://reviewable.io/reviews/starkware-libs/sequencer/2845#-OEwySC4C83GIQENYvEo:-OExUU5O8JTif_MY2d9G:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/8d13ce882fac7aa074a12081b03d1070633f4694/crates/papyrus_base_layer/src/ethereum_base_layer_contract.rs#L96)):*
<details><summary><i>Previously, matan-starkware wrote…</i></summary><blockquote>

Mention the try and error stuff it basic rust (we can see TryInto).
</blockquote></details>

Done.

**Review by elintul — APPROVED** [review]
> Reviewed 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2845)* status: all files reviewed, 4 unresolved discussions (waiting on @giladchase, @graphite-app[bot], and @matan-starkware)

___
*[`crates/papyrus_base_layer/src/ethereum_base_layer_contract.rs` line 95 at r5](https://reviewable.io/reviews/starkware-libs/sequencer/2845#-OF0xDt52jbm2Sqqw13P:-OF0xDt52jbm2Sqqw13Q:b-mjzxqv) ([raw file](https://github.com/starkware-libs/sequencer/blob/4cb4ffba4583f614df55f564ff87c887a7660967/crates/papyrus_base_layer/src/ethereum_base_layer_contract.rs#L95)):*
> ```XML
> 
>         let matching_logs = self.contract.provider().get_logs(&filter).await?;
>         // Convert logs to L1Events.
> ```

I'd refrain from using struct/enum names in doc.; it's a false-positive when searching the codebase.



_Suggestion:_
```XML
L1 events
```

**Review by giladchase — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2845)* status: all files reviewed, 4 unresolved discussions (waiting on @elintul, @graphite-app[bot], and @matan-starkware)

___
*[`crates/papyrus_base_layer/src/ethereum_base_layer_contract.rs` line 95 at r5](https://reviewable.io/reviews/starkware-libs/sequencer/2845#-OF0xDt52jbm2Sqqw13P:-OFI32ej5E0W6whg8l_Y:btvalvi) ([raw file](https://github.com/starkware-libs/sequencer/blob/4cb4ffba4583f614df55f564ff87c887a7660967/crates/papyrus_base_layer/src/ethereum_base_layer_contract.rs#L95)):*
<details><summary><i>Previously, elintul (Elin) wrote…</i></summary><blockquote>

I'd refrain from using struct/enum names in doc.; it's a false-positive when searching the codebase.
</blockquote></details>

Yr right, done.

**Review by elintul — APPROVED** [review]
> Reviewed 2 of 2 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2845)* status: all files reviewed, 3 unresolved discussions (waiting on @graphite-app[bot] and @matan-starkware)

**Review by matan-starkware — APPROVED** [review]
> Reviewed 2 of 2 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2845)* status: all files reviewed, 1 unresolved discussion (waiting on @graphite-app[bot])

**Review by matan-starkware — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2845)* status: all files reviewed, 2 unresolved discussions (waiting on @giladchase and @graphite-app[bot])

___
*[`crates/papyrus_base_layer/src/ethereum_base_layer_contract.rs` line 96 at r6](https://reviewable.io/reviews/starkware-libs/sequencer/2845#-OFIWEvx-YImUd7Y93h1:-OFIWEvx-YImUd7Y93h2:b-ull51s) ([raw file](https://github.com/starkware-libs/sequencer/blob/cd1938044bf120776cea23422c37eca5ad5b437f/crates/papyrus_base_layer/src/ethereum_base_layer_contract.rs#L96)):*
> ```XML
>         let matching_logs = self.contract.provider().get_logs(&filter).await?;
>         // Convert logs to L1 events.
>         matching_logs.into_iter().map(TryInto::try_into).collect()
> ```

Honestly, given the variable name and the function return type idk if we need any comment.



_Suggestion:_
```XML
        matching_logs.into_iter().map(TryInto::try_into).collect()
```

**Review by giladchase — COMMENTED** [review]
> Reviewed 1 of 6 files at r1, 1 of 5 files at r2, 2 of 4 files at r3, 1 of 3 files at r4, 1 of 2 files at r6, 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2845)* status: all files reviewed, 1 unresolved discussion (waiting on @graphite-app[bot])

**Review by giladchase — COMMENTED** [review]
> Dismissed @graphite-app[bot] from a discussion.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2845)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @giladchase)
